replica/table.cc

```diff
@@ -1838,6 +1840,33 @@ void table::update_effective_replication_map(locator::effective_replication_map_
     if (old_erm) {
         old_erm->invalidate();
     }
+
+    _stats.tablet_count_on_current_shard = calculate_tablet_count_on_current_shard();
+}
+
+int64_t table::calculate_tablet_count_on_current_shard() const {
+    if (!uses_tablets()) {
+        return 0;
+    }
+
+    const auto& token_metadata = _erm->get_token_metadata();
+    const auto& tablet_map = token_metadata.tablets().get_tablet_map(_schema->id());
+    const auto this_host_id = token_metadata.get_topology().my_host_id();
+
+    int64_t new_tablet_count_on_this_shard{0};
+
+    for (auto tablet_id : tablet_map.tablet_ids()) {
+        const std::optional<shard_id> shard_id = tablet_map.get_shard(tablet_id, this_host_id);
+        if (!shard_id.has_value()) {
+            continue;
+        }
```

Comment:
We should revisit this after https://github.com/scylladb/scylladb/issues/16877, and just count non-null storage groups.